<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="rviz" default="true" />
  <arg name="car_name" default="dr_drift" />
  <!-- Launch car_state_node -->
  <node name="car_state" ns="simulation" pkg="gazebo_simulation" type="car_state_node" output="screen">
    <param name="car_name" value="$(arg car_name)" />
    <rosparam file="$(find gazebo_simulation)/param/car_state/default.yaml" command="load" />
    <rosparam file="$(find gazebo_simulation)/param/car_state/topics.yaml" command="load" ns="topics" />
    <rosparam file="$(find gazebo_simulation)/param/model_plugin_link/topics.yaml" command="load" ns="topics/model_plugin" />
  </node>
  <!-- Launch additional visualization node -->
  <node if="$(arg rviz)" name="car_state_visualization_node" ns="gazebo_simulation" pkg="gazebo_simulation" type="car_state_visualization_node" output="screen">
    <rosparam file="$(find gazebo_simulation)/param/car_state/default.yaml" command="load" />
    <rosparam file="$(find gazebo_simulation)/param/car_state/topics.yaml" command="load" ns="topics" />
  </node>
</launch>
